Src/IronPython/Runtime/Operations/ArrayOps.cs

```diff
@@ -180,7 +179,9 @@ public static class ArrayOps {
             if (a.Rank != args.Length) throw PythonOps.ValueError("bad dimensions for array, got {0} expected {1}", args.Length, a.Rank);
 
             for (int i = 0; i < indices.Length; i++) indices[i] += a.GetLowerBound(i);
-            a.SetValue(indexAndValue[indexAndValue.Length - 1], indices);
+
+            Type elm = a.GetType().GetElementType()!;
```

Comment:
Minor nit: could use `t.GetElementType()` (or we could get rid of `t`).

Reply:
Oh yeah, indeed. I've just mindlessly copied the line from the next overload (got distracted by error handling here; see incoming issue report)